src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemAlignmentUtils.java

```diff
@@ -57,9 +67,15 @@ public static SAMRecord applyAlignment( final String readName, final byte[] base
                         }
                     }
                 }
-                bases = Arrays.copyOfRange(bases, alignment.getSeqStart(), alignment.getSeqEnd());
+            }
+            if (tmpCigar.getFirstCigarElement().getOperator() == CigarOperator.H ||
+                    tmpCigar.getLastCigarElement().getOperator() == CigarOperator.H) {
+                final int leftHardClip = tmpCigar.getFirstCigarElement().getOperator() == CigarOperator.H ? tmpCigar.getFirstCigarElement().getLength() : 0;
```

Comment:
There still a diff between soft and hard-clipping, where the the soft determine the positions of the bases[] array relative to the full read sequence. The accessor names do not rule that possibility. I just wanted to make sure. 

The reason the old more succinct code works is because it only get executed with that ```if``` statement where is certain that all clipping is to be excised, however that is not in general true any longer if we are to support those "invalid" inputs. That is the reason for the more complex code, unless I have not realized of something.